metadata-models/src/main/resources/entity-registry.yml

```diff
@@ -419,6 +419,19 @@ entities:
     keyAspect: dataHubViewKey
     aspects:
       - dataHubViewInfo
+  - name: join
+    doc: Join Dataset Fields
+    keyAspect: joinKey
+    aspects:
```

Comment:
Is there a reason why all these aspects are needed? `joinProperties`, `editableJoinProperties`, and maybe `status` are the only that seem necessary to me, but would love to hear the rationale for the other ones

Reply:
InstitutionalMemory makes sense to remove. What would BrowsePaths look like if the underlying datasets have two different BrowsePaths?

Reply:
After discussion with the teams, InstitutionalMemory is needed.  It is a place where documentation links can be provided.

BrowsePaths are a way to organize Joins.  Since Joins are completely different entities, their browse paths are different from datasets.